src/rdkafka_txnmgr.c

```diff
@@ -2390,6 +2413,8 @@ rd_kafka_txn_op_commit_transaction_ack(rd_kafka_t *rk,
                                        rd_kafka_q_t *rkq,
                                        rd_kafka_op_t *rko) {
         rd_kafka_error_t *error;
+        rd_bool_t complete    = rd_true;
```

Comment:
..transaction_ack() should be called when everything is done, the draining must be performed in an earlier stage before we transition not .._NOT_ACKED.